### PR TITLE
Fix ceph recipe

### PR DESCRIPTION
**Pull request by vuntz**

### CODE REVIEW

File: chef/cookbooks/glance/recipes/ceph.rb [hunk 1]
```diff
@@ -17,38 +17,47 @@
...
+  # Ensure that the ceph config file that will be written in the glance config
+  # file is the right one
+  node.default[:glance][:rbd][:store_ceph_conf] = ceph_conf
 else
-  # If Ceph confifuration file is present,
-  # external Ceph cluster will be used,
+  ceph_conf = node[:glance][:rbd][:store_ceph_conf]
+  admin_keyring = node[:glance][:rbd][:store_admin_keyring]
+
+  # If Ceph confifuration file is present, external Ceph cluster will be used,
```
Comment (rhafer) [inline chef/cookbooks/glance/recipes/ceph.rb]: You might wanna fix that "confifuration" typo as well :)

File: chef/cookbooks/glance/recipes/ceph.rb [hunk 2]
```diff
@@ -17,38 +17,47 @@
...
 ceph_env_filter = " AND ceph_config_environment:ceph-config-default"
 ceph_servers = search(:node, "roles:ceph-osd#{ceph_env_filter}") || []
 if ceph_servers.length > 0
   include_recipe "ceph::keyring"
+
+  ceph_conf = "/etc/ceph/ceph.conf"
+  admin_keyring = "/etc/ceph/ceph.client.admin.keyring"
+  # Ensure that the ceph config file that will be written in the glance config
+  # file is the right one
+  node.default[:glance][:rbd][:store_ceph_conf] = ceph_conf
```
Comment (rhafer) [inline chef/cookbooks/glance/recipes/ceph.rb]: I might be confused, but does this really have the desired effect? Doesn't the role attribute that's set by the glance proposal have a higher precedence than a default node attribute? (At least that's what the chef docs say)

Comment (vuntz) [inline chef/cookbooks/glance/recipes/ceph.rb]: No, the attributes from the proposal are copied at the default level. See https://github.com/crowbar/barclamp-crowbar/blob/master/crowbar_framework/app/models/service_object.rb#L1060

File: chef/cookbooks/glance/recipes/ceph.rb (hunk 1 above)
Comment (vuntz) [inline chef/cookbooks/glance/recipes/ceph.rb]: Heh, fixed

File: chef/cookbooks/glance/recipes/ceph.rb (hunk 2 above)
Comment (rhafer) [inline chef/cookbooks/glance/recipes/ceph.rb]: Right, as default __role__ attributes. And according to http://docs.chef.io/attributes.html#attribute-precedence those have a higher precedence than default attributes set in a recipe. But I am sure I am overlooking something here ;)

Comment (vuntz) [inline chef/cookbooks/glance/recipes/ceph.rb]: Oh, hrm. I don't know, then :-) I guess we'll see how it breaks if someone decides to trigger this nice edge case ;-)

I'll try to setup a cloud and trigger this code, to see how it goes.

Comment (vuntz) [inline chef/cookbooks/glance/recipes/ceph.rb]: So I think it works fine: I had an attribute with a bogus value in the proposal, and in the end, I have the correct value in glance-api.conf.

Comment (vuntz) [inline chef/cookbooks/glance/recipes/ceph.rb]: Hrm, downside, though, is that if I change the attribute in the proposal, the new value will never be used...

Comment (mjura) [inline chef/cookbooks/glance/recipes/ceph.rb]: This ok, if we will decide that Ceph deployed by Crowbar has higher priority.
Comment from my site is that we have to add also this line too
```
node.default[:glance][:rbd][:store_admin_keyring] = admin.keyring
```

Comment (mjura) [inline chef/cookbooks/glance/recipes/ceph.rb]: I've checked this and works fine, except missing line from above

Comment (vuntz) [inline chef/cookbooks/glance/recipes/ceph.rb]: > Comment from my site is that we have to add also this line too
> ```
> node.default[:glance][:rbd][:store_admin_keyring] = admin.keyring
> ```

Why? Is the attribute used elsewhere? The reason I do it for the conf file is that the attribute is being used in glance-api.conf.

Comment (mjura) [inline chef/cookbooks/glance/recipes/ceph.rb]: Ok, I've checked this, you are right.

**Comment by mjura** [comment]
It was tested :+1:

**Comment by tserong** [comment]
+1

**Comment by jsuchome** [comment]
Looks good

**Pull request merged**